vault/generate_root.go

```diff
@@ -111,30 +127,41 @@ func (c *Core) GenerateRootConfiguration() (*GenerateRootConfig, error) {
 		return nil, consts.ErrStandby
 	}
 
-	c.generateRootLock.Lock()
-	defer c.generateRootLock.Unlock()
+	c.namespaceRootGenLock.Lock()
+	defer c.namespaceRootGenLock.Unlock()
 
-	// Copy the config if any
-	var conf *GenerateRootConfig
-	if c.generateRootConfig != nil {
-		conf = new(GenerateRootConfig)
-		*conf = *c.generateRootConfig
-		conf.OTP = ""
-		conf.Strategy = nil
+	if c.namespaceRootGens[ns.UUID] == nil {
+		return nil, nil
 	}
-	return conf, nil
+
+	config := c.namespaceRootGens[ns.UUID].Config.Copy()
```

Comment:
```suggestion
	config := *c.namespaceRootGens[ns.UUID].Config
```

and then you can spare the `func (config GenerateRootConfig) Copy() GenerateRootConfig`.

Alternatively a dedicated `Copy()` might make more sense if you also include the

```go
config.OTP = ""
config.Strategy = nil
```

steps in it, sth like this:

```go
func (config GenerateRootConfig) Copy() GenerateRootConfig {
	config.OTP = ""
    config.Strategy = nil
	return config
}
```